sql/core/src/main/scala/org/apache/spark/sql/expressions/UserDefinedFunction.scala

```diff
@@ -103,4 +110,19 @@ case class UserDefinedFunction protected[sql] (
       udf
     }
   }
+
+  /**
+   * Updates UserDefinedFunction to non-deterministic.
+   *
+   * @since 2.3.0
+   */
+  def nonDeterministic(): UserDefinedFunction = {
```

Comment:
shall we follow existing naming style and make it `def withDeterministic(deterministic: Boolean)`?